pkgs/tools/typesetting/tex/texlive/combine.nix

```diff
@@ -57,54 +78,40 @@ in (buildEnv {
   passthru.packages = pkgList.all;
 
   postBuild = ''
-    mkdir -p "$out"/bin
-  '' +
-    lib.concatMapStrings
-      (path: ''
-        for f in '${path}'/bin/*; do
-          if [[ -L "$f" ]]; then
-            cp -d "$f" "$out"/bin/
-          else
-            ln -s "$f" "$out"/bin/
-          fi
-        done
-      '')
-      pkgList.bin
-    +
-  ''
-    export PATH="$out/bin:$out/share/texmf/scripts/texlive:$PATH"
-    export TEXMFCNF="$out/share/texmf/web2c"
+    TEXMFDIST="${texmf}/share/texmf"
+    export PATH="$out/bin:$TEXMFDIST/share/texmf/scripts/texlive:$PATH"
+    export PERL5LIB="$TEXMFDIST/share/texmf/scripts/texlive:${bin.core.out}/share/texmf-dist/scripts/texlive"
     TEXMFSYSCONFIG="$out/share/texmf-config"
     TEXMFSYSVAR="$out/share/texmf-var"
-    export PERL5LIB="$out/share/texmf/scripts/texlive:${bin.core.out}/share/texmf-dist/scripts/texlive"
+    export TEXMFCNF="$TEXMFSYSVAR/web2c"
```

Comment:
I am fairly sure it's ok. Kpathsea will read all `texmf.cnf` files it can find, but apply the settings in order of priority, with `TEXMFCNF` coming first. So the parts we change will apply, and if we forget to copy some file, it will still be read from `TEXMFDIST`.

(To be sure, I checked the documentation right now: https://www.tug.org/texinfohtml/kpathsea.html#Config-files)

Reply:
PS: on second though, the behaviour of Kpathsea means we can simply write the relevant variables in a new empty `texmf.cnf`, instead of using sed and regular expressions. But I was trying to keep this PR on point – this is rather something to try and test carefully in the future.

Reply:
Note for the future: avoid creating an entire copy of `texmf.cnf{.lua}` and just override the variables that need overriding. I'd prefer not to do this here as I'd prefer to test it separately, after the split is proven to work correctly.